uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2RepoServiceImpl.java

```diff
@@ -232,20 +233,17 @@ GAV getGAV(final String path) {
     }
 
     /**
-     * @param baseURL the base URL where Guvnor M2 repo is hosted in web container. return a Guvnor M2 repo
-     * URL point to local file system if baseURL is not available.
+     * URL point to local file system if URL property is not available.
      * @return String
      */
     @Override
-    public String getRepositoryURL(final String baseURL) {
-        if (baseURL == null || baseURL.isEmpty()) {
+    public String getRepositoryURL() {
+        final String urlProperty = System.getProperty(ArtifactRepositoryService.GLOBAL_M2_REPO_URL);
+
+        if (urlProperty == null || urlProperty.isEmpty()) {
             return repository.getRepositoryURL(ArtifactRepositoryService.GLOBAL_M2_REPO_NAME);
         } else {
-            if (baseURL.endsWith("/")) {
-                return baseURL + "maven2/";
-            } else {
-                return baseURL + "/maven2/";
-            }
+            return urlProperty;
```

Comment:
Would it make sense to add some error checking, it the string from the system property constitutest valid url?
Like createing `java.net.URL(String)` and catching `MalformedUrlException` and then either rethrowing or returning some default..

Reply:
Sure we can do that.